overlord/snapstate/handlers.go

```diff
@@ -997,7 +1036,73 @@ func (m *SnapManager) undoRefreshAliasesV2(t *state.Task, _ *tomb.Tomb) error {
 
 	snapst.AutoAliasesDisabled = autoDisabled
 	snapst.Aliases = oldAliases
-	Set(st, snapName, snapst)
+	newSnapStates := make(map[string]*SnapState, 1+len(otherSnapDisabled))
+	newSnapStates[snapName] = snapst
+
+	// if we disabled other snap aliases try to undo that
+	var conflicting map[string]bool
+	var curOtherSnapStates map[string]*SnapState
+	if len(otherSnapDisabled) != 0 {
+		conflicting = make(map[string]bool, len(otherSnapDisabled))
+		curOtherSnapStates = make(map[string]*SnapState, len(otherSnapDisabled))
+		for otherSnap, otherDisabled := range otherSnapDisabled {
+			var otherSnapSt SnapState
+			err := Get(st, otherSnap, &otherSnapSt)
+			if err != nil {
+				return err
+			}
+			curOtherInfo, err := otherSnapSt.CurrentInfo()
+			if err != nil {
+				return err
+			}
+
+			curOtherSnapStates[otherSnap] = &otherSnapSt
+
+			autoDisabled := otherSnapSt.AutoAliasesDisabled
+			if otherDisabled.Disabled {
```

Comment:
I found this one a bit confusing, because there are two possible meanings to the flag:
* The AutoAliasesDisabled flag was true before
* The AutoAliasesDisabled flag was made true

Turns out the meaning is the second one, which was slightly surprising because the Manual field as well as other similar undo-state data tend to hold the old state for reversal. Also, this flag is only about the Auto flags and it won't be affected by manual removals, right? 

Perhaps naming it AutoDisabled and inverting the semantic would clarify things slightly.